packages/proxy/lib/http/util/service-worker-injector.ts

```diff
@@ -30,20 +43,15 @@ declare let self: ServiceWorkerGlobalScope
 export const injectIntoServiceWorker = (body: Buffer) => {
   function __cypressInjectIntoServiceWorker () {
     let listenerCount = 0
-    let eventQueue: ServiceWorkerClientEvent[] = []
     const nonCaptureListenersMap = new WeakMap<EventListenerOrEventListenerObject, EventListenerOrEventListenerObject>()
     const captureListenersMap = new WeakMap<EventListenerOrEventListenerObject, EventListenerOrEventListenerObject>()
     const targetToWrappedHandleEventMap = new WeakMap<Object, EventListenerOrEventListenerObject>()
     const targetToOrigHandleEventMap = new WeakMap<Object, EventListenerOrEventListenerObject>()
 
-    const sendEvent = (event: ServiceWorkerClientEvent) => {
-      // if the binding has been created, we can call it
-      // otherwise, we need to queue the event
-      if (self.__cypressServiceWorkerClientEvent) {
-        self.__cypressServiceWorkerClientEvent(JSON.stringify(event))
-      } else {
-        eventQueue.push(event)
-      }
+    const sendEvent = (event: ServiceWorkerClientEventWithoutScope) => {
+      const payload = Object.assign({}, event, { scope: self.registration.scope })
+
+      self.__cypressServiceWorkerClientEvent!(JSON.stringify(payload))
```

Comment:
Do we not need the queue anymore?

Reply:
We are now waiting during the install event for the binding to be available so the queue is no longer needed.